src/DependencyEmitter/BasicDependencyEmitter.php

```diff
@@ -33,6 +33,42 @@ public function supportsParser(AstParserInterface $astParser)
         return $astParser instanceof NikicPhpParser;
     }
 
+    public function applyDependencies(
+        AstParserInterface $astParser,
+        AstMap $astMap,
+        DependencyResult $dependencyResult
+    ) {
+        /** @var $astParser NikicPhpParser */
```

Comment:
why did you removed the assert?

Reply:
Because not all emitters have used this. But I added a meaningful exception class instead.